src/free.rs

```diff
@@ -14,8 +14,8 @@ use alloc::{
     string::String,
 };
 
-#[cfg(feature = "use_alloc")]
```

Comment:
I'm no expert on `cfg`-stuff. Could you elaborate why this was removed?

Reply:
`intersperse` does not need `use_alloc` (works in `no_std` mode). This `#[cfg(feature = "use_alloc")]` was likely added here to avoid an `unused_imports` warning during `cargo check --no-default-features`  – since everything else in `free.rs` that depends on `Itertools` requires `use_alloc`. Since now `intersperse` is here, `Itertools` needs to be in scope when no features are set.